mlflow/deployments/base.py

```diff
@@ -70,6 +70,213 @@ def target_help():
+
+        :param deployment: Name of deployment to update
+        :param endpoint: Name of the endpoint containing the deployment to update
+        :param model_uri: URI of a new model to deploy.
+        :param flavor: (optional) new model flavor to use for deployment. If provided,
+                       ``model_uri`` must also be specified. If ``flavor`` is unspecified but
+                       ``model_uri`` is specified, a default flavor will be chosen and the
+                       deployment will be updated using that flavor.
+        :param config: (optional) dict containing updated target-specific configuration for the
+                       deployment
+        :return: None
+        """
+        pass
+
+    @abc.abstractmethod
+    def delete_deployment(self, deployment, endpoint):
+        """
+        Delete the deployment with name ``name`` from the specified endpoing. Deletion should be
+        idempotent (i.e. deletion should not fail if retried on a non-existent deployment).
+
+        :param deployment: Name of deployment to delete
+        :param endpoint: Name of the Endpoint containing the deployment to delete
+        :return: None
+        """
+        pass
+
+    @abc.abstractmethod
+    def list_deployments(self, endpoint):
+        """
+        List deployments within the specified endpoint. This method is expected to return
+        an unpaginated list of all deployments (an alternative would be to return a dict
+        with a 'deployments' field containing the actual deployments, with plugins able to
+        specify other fields, e.g. a next_page_token field, in the returned dictionary for
+        pagination, and to accept a `pagination_args` argument to this method for passing
+        pagination-related args).
+
+        :param endpoint: Name of the Endpoint to list all contained deployments
+        :return: A list of dicts corresponding to deployments. Each dict is guaranteed to
+                 contain a 'name' key containing the deployment name. The other fields of
+                 the returned dictionary and their types may vary across deployment targets.
+        """
+        pass
+
+    @abc.abstractmethod
+    def get_deployment(self, deployment, endpoint):
+        """
+        Returns a dictionary describing the specified deployment, throwing a
+        py:class:`mlflow.exception.MlflowException` if no deployment exists with the provided
+        name in the provided endpoint.
+        The dict is guaranteed to contain an 'name' key containing the deployment name.
+        The other fields of the returned dictionary and their types may vary across
+        deployment targets.
+
+        :param deployment: Name of deployment to fetch
+        :param endpoint: Name of the Endpoint containing the deployment to fetch
+        """
+        pass
+
+    @abc.abstractmethod
+    def predict(self, endpoint, df, deployment=None):
```

Comment:
Regarding the extensibility of `BaseDeploymentClient.predict()` (https://github.com/databricks/universe/pull/163268) to endpoints, I think the following signature should work:

```
def predict(self, deployment=None, endpoint=None):
     """
     Exactly one of `deployment` or `endpoint` must be specified.
     """
```

Consider preexisting code that calls `predict("my_deployment")`. This will continue to behave as it did before, since `deployment` remains the first positional arg. New code for predicting on an endpoint can be constructed as `predict(endpoint="my_endpoint").`